langstream-agents/langstream-agents-flow-control/src/main/java/ai/langstream/agents/flow/DispatchAgent.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.langstream.agents.flow;
+
+import ai.langstream.ai.agents.commons.TransformContext;
+import ai.langstream.ai.agents.commons.jstl.predicate.JstlPredicate;
+import ai.langstream.api.runner.code.AbstractAgentCode;
+import ai.langstream.api.runner.code.AgentContext;
+import ai.langstream.api.runner.code.AgentProcessor;
+import ai.langstream.api.runner.code.Record;
+import ai.langstream.api.runner.code.RecordSink;
+import ai.langstream.api.runner.topics.TopicProducer;
+import ai.langstream.api.runtime.ComponentType;
+import ai.langstream.api.util.ConfigurationUtils;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DispatchAgent extends AbstractAgentCode implements AgentProcessor {
+
+    record Route(String destination, JstlPredicate predicate) {}
+
+    private final List<Route> routes = new ArrayList<>();
+    private final Map<String, TopicProducer> producers = new HashMap<>();
+    private AgentContext agentContext;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void init(Map<String, Object> configuration) {
+        List<Map<String, Object>> routes =
+                (List<Map<String, Object>>) configuration.getOrDefault("routes", List.of());
+        routes.forEach(
+                r -> {
+                    String when = ConfigurationUtils.getString("when", "", r);
+                    String destination = ConfigurationUtils.getString("destination", "", r);
```

Comment:
what about using another param "drop" ? 
It could be error-prone this configuration

so then either "destination" or "drop" are required and you can fail the deploy

Reply:
having two different options may make difficult to use a variable for this value, think about a configurable application with global variables.